FutureNorthants/northants-design-system
# Update to axios 1, but make it peer dependency

Fixes #355 

This PR upgrades Axios to the latest version, but it moves it from a dependency to a dev dependency and also adds it as a peer dependency. Without these changes the frontend doesn't work and returns `TextEncoder is not a function`. From what I can tell, it looks like rollup packages axios as a node app so it won't work in the browser. 

**This means axios is excluded from rollup and needs to be added manually to the frontend as a dependency.**

## Testing
- Checkout this branch and run `npm install` then `npm run dev` to manually test the PostcodeSearch component
- Run `yalc publish` then in the frontend, run `yalc add northants-design-system` and `yarn install`
- Then add `yarn add axios` to the frontend, before running `yarn dev` to test the PostcodeSearch on a page with signposts still works as expected. 
- Run `yarn build` to test the frontend builds correctly
